ansible/roles/iroha-docker/tasks/config-gen.yml

```diff
@@ -1,63 +1,87 @@
+            IROHA_ACCOUNT_KEYS: "{{ hostvars[iroha_command_host]['iroha_service_account_keys'] | join(',') }}"
+        register: iroha_command_register
 
-  - name: cleanup
-    file:
-      path: "{{ iroha_config_local_dir }}/{{ item[0].hostname }}{{ item[1] }}"
-      state: absent
-    loop: "{{ iroha_nodes | product(['.priv', '.pub']) | list }}"
+      - name: Get existing genesis block 2
+        set_fact:
+          iroha_new_genesis_block: "{{ (iroha_command_register.ansible_facts.docker_container.Output | from_json).result }}"
 
-  - name: cleanup
-    file:
-      path: "{{ iroha_config_local_dir }}/genesis.block"
-      state: absent
+    delegate_to: "{{ iroha_command_host }}"
     run_once: yes
-  become: no
-  delegate_to: localhost
-  when: iroha_generate_init_configs
-  tags: ["iroha-docker", "iroha-config-gen"]
+    when: node_change
 
-- name: create deploy dir
-  file:
-    state: directory
-    path: "{{ iroha_deploy_dir }}/conf"
-  tags: ["iroha-docker", "iroha-config-gen"]
+  - block:
+
+      - name: Generate genesis.block 1
+        set_fact:
+          iroha_new_genesis_block: "{{ iroha_genesis_block | from_json }}"
+
+      - name: Generate genesis.block 2
+        set_fact:
+          iroha_genesis_block_commands: "{{ (iroha_genesis_block_commands | default([])) + [item] }}"
+        loop: "{{ iroha_new_genesis_block['block_v1']['payload']['transactions'][0]['payload']['reducedPayload']['commands'] }}"
+        when: "{{ 'addPeer' not in item }}"
+
+      - name: Generate genesis.block 3
+        set_fact:
+          iroha_genesis_block_commands: "{{ iroha_genesis_block_commands + [{'addPeer': {'peer': {'address': item.hostname, 'peerKey': item.pub_key}}}] }}"
+        loop: "{{ iroha_all_new_nodes }}"
+
+      - name: Generate genesis.block 4
+        set_fact:
+          iroha_new_genesis_block: "{{ iroha_new_genesis_block | combine({'block_v1': {'payload': {'transactions': [{'payload': {'reducedPayload': {'commands': iroha_genesis_block_commands}}}]}}}, recursive=True) }}"
+
+    delegate_to: localhost
+    run_once: yes
+    become: no
+    when: not node_change and iroha_all_new_nodes
+
+  - name: Debug
+    debug:
+      var: iroha_new_genesis_block
+    run_once: yes
+    become: no
+    delegate_to: localhost
+
+  - name: Move genesis.block
+    copy:
+      content: "{{ iroha_new_genesis_block | to_nice_json(indent=2) }}"
+      dest: "{{ iroha_deploy_dir }}/conf/{{ item.human_hostname }}/genesis.block"
+    loop: "{{ iroha_new_nodes }}"
+
+  - name: Move pub peer keys
+    copy:
+      content: "{{ iroha_new_peer_keys[item.hostname]['pub'] }}"
+      dest: "{{ iroha_deploy_dir }}/conf/{{ item.human_hostname }}/{{ item.hostname }}.pub"
+    loop: "{{ iroha_new_nodes }}"
+
+  - name: Move priv peer keys
+    copy:
+      content: "{{ iroha_new_peer_keys[item.hostname]['priv'] }}"
+      dest: "{{ iroha_deploy_dir }}/conf/{{ item.human_hostname }}/{{ item.hostname }}.priv"
+    loop: "{{ iroha_new_nodes }}"
```

Comment:
why you run it in loop? Do all nodes contain all private key

Reply:
Because this step must have "human_hostname" for path generation for each node, the "iroha_new_nodes" variable contains it, but the "iroha_new_peer_keys" variable doesn't contain it.